cmd/serverless/main.go

```diff
@@ -250,49 +251,47 @@ func runAgent(ctx context.Context, stopCh chan struct{}) (err error) {
 		log.Error("No API key configured, exiting")
 	}
 
-	// starts logs collection if enabled
-	// ---------------------------------
-
-	if config.Datadog.GetBool("logs_enabled") {
-
-		// type of logs we are subscribing to
-		var logsType []string
-		// TODO(remy): there is two different things that we may have to differentiate:
-		//             the user may want to send these logs to the intake, and the Agent
-		//             may want to fetch more than what he want sent. For instance, if the
-		//             user don't care about the platform logs, the Agent will still want
-		//             to receive them in order to generate the enhanced metrics.
-		if envLogsType, exists := os.LookupEnv(logsLogsTypeSubscribed); exists {
-			parts := strings.Split(strings.TrimSpace(envLogsType), " ")
-			for _, part := range parts {
-				part = strings.ToLower(strings.TrimSpace(part))
-				switch part {
-				case "function", "platform", "extension":
-					logsType = append(logsType, part)
-				default:
-					log.Warn("While subscribing to logs, unknown log type", part)
-				}
+	// restore the current function ARN and request ID from the cache in case the extension was restarted
+	// ---------------------------
+
+	aws.RestoreCurrentARNFromCache()
+	aws.RestoreCurrentRequestIDFromCache()
+
+	// starts logs collection
+	// ----------------------
+
+	// type of logs we are subscribing to
+	var logsType []string
+	if envLogsType, exists := os.LookupEnv(logsLogsTypeSubscribed); exists {
```

Comment:
While previously thinking about this feature, I was thinking about it that way:
  1. store here the configuration of which types of logs we would send to the intake
  2. subscribe to what have been configured *but always include the platforms one*
  3. send only the ones matching the types stored in (1)
This way, we would always have the platform logs and thus, we would always be able to generate the enhanced metrics, but they may not be always sent to the intake.

We're very close to this with your implementation, and I 100% get what you're achieving here. Your implementation collect and send exactly what is configured in `logsLogsTypeSubscribed` which is great, but it also means there is a scenario where we would not have the platform logs if someone configures it this way (which is OK, as long as the person knows what will happen with enhanced metrics).

I'm curious what do you think about this?

Reply:
Ah, that makes sense. I think my proposed solution would be to **always** subscribe to **all** logs, and then move the logic that filters them based on the environment variable entirely to the processing code. So the environment variable would control what gets sent to the intake, not what we are subscribed to.

Do you think it would be a problem to subscribe to function/extension logs even if we aren't going to send them all? If it is not acceptable, we could also implement a solution where we subscribe to **at least** platform logs (plus others specified), and then **also** do a filtering pass when we send to the intake. That would be slightly more complicated but should also work fine.

Either way, I think this could be handled in a subsequent PR, it's not directly related to the changes around timeouts and the behavior of the LOGS_ENABLED env var.

Reply:
> Do you think it would be a problem to subscribe to function/extension logs even if we aren't going to send them all?

In all honesty I don't have a strong opinion, both would technically work very well, that's why I've started the convo :D. Time measurements will presumably help decide what's the best here but you're right, this should go in another PR anyway. 👍